cidr.go

```diff
@@ -127,6 +127,22 @@ func SplitIPNetIntoN(iprange *net.IPNet, n int) ([]*net.IPNet, error) {
 	return subnets, nil
 }
 
+// DifferenceCIDR set difference of target and exclude
+func DifferenceCIDR(target, exclude *net.IPNet) []*net.IPNet {
+	left, _, right := partitionCIDR(*target, *exclude)
+	return append(left, right...)
+}
+
+// ContainsCIDR check if A contains B
+func ContainsCIDR(A, B *net.IPNet) bool {
+	Bfirst, Blast, _ := AddressRange(B)
+	if A.Contains(Bfirst) && A.Contains(Blast) {
```

Comment:
I think we can use:
```go
return A.Contains(Bfirst) && A.Contains(Blast)
```

Reply:
No, Consider two examples.

### 1. 
If ip is less than cdn, like.

ip.txt: `1.1.1.1/28`

cdn.txt: `1.1.1.1/24`

no output, `return A.Contains(Bfirst) && A.Contains(Blast)`  is correct,

### 2.
but If If ip is greater than cdn.

ip.txt: `1.1.1.1/20`

cdn.txt: `1.1.1.1/24`

We need to subtract /24 from /20.
output:
```
1.1.0.0/24
1.1.8.0/21
1.1.4.0/22
1.1.2.0/23
```

Reply:
If I'm not mistaken, the output of the following code snippet:
```go
if A.Contains(Bfirst) && A.Contains(Blast) { // true if both conditions are matched
	return true // both conditions matched
} else {
	return false
}
```
is equivalent to directly return the result of the boolean comparison:
```go
return A.Contains(Bfirst) && A.Contains(Blast)
```

Reply:
You are right. I misunderstood the code snippet